app/src/main/java/com/example/composetemplate/repositories/FirebaseActionable.kt

```diff
@@ -98,26 +148,91 @@ class FirebaseActionable : AuthActionable, TokenFetcher {
         }
     }
 
-    /** Parse the expiration date from token in firebase mode.*/
+    /**
+     * Parses the expiration date from the Firebase ID token.
+     *
+     * @param token The Firebase ID token as a String.
+     * @return The expiration time of the token in milliseconds.
+     */
     private fun parseExpirationDate(token: String?): Long {
         return try {
             token ?: return 0L
-            // Split the token into parts: header, payload, and signature
+            // Split the token into its components (header, payload, signature)
             val parts = token.split(".")
             if (parts.size != 3) {
                 return 0L // Invalid token format
             }
 
-            // Decode the payload part from Base64
+            // Decode the payload from Base64 URL-safe encoding
             val payload = String(Base64.decode(parts[1], Base64.URL_SAFE))
             // Parse the payload as a JSON object
             val jsonObject = JSONObject(payload)
-            // Get the expiration time (exp) from the JSON object (convert seconds to milliseconds)
+            // Extract the expiration time (exp) and convert from seconds to milliseconds
             jsonObject.optLong("exp", 0) * 1000
         } catch (e: Exception) {
             e.printStackTrace()
-            0L
+            0L // Return 0 if parsing fails
         }
     }
 
+    /**
+     * Performs Google Sign-In using Google Identity Services and Firebase.
+     *
+     * @param googleCredentialAuthParameter Parameters needed for Google Sign-In, including the activity context.
+     * @return A [SignInResult] indicating the outcome of the sign-in operation.
+     */
+    override suspend fun googleSignIn(googleCredentialAuthParameter: GoogleCredentialAuthParameter): SignInResult {
```

Comment:
The function is a bit long. Would it be better to break it into smaller functions or create inner functions?